rest/server_context.go

```diff
@@ -924,11 +924,12 @@ func (sc *ServerContext) asyncDatabaseOnline(nonCancelCtx base.NonCancellableCon
 		return
 	}
 
-	base.DebugfCtx(ctx, base.KeyConfig, "Async database initialization complete, starting online processes...")
+	base.InfofCtx(ctx, base.KeyAll, "Async database initialization complete, starting online processes...")
 	err := dbc.StartOnlineProcesses(ctx)
 	if err != nil {
-		base.InfofCtx(ctx, base.KeyAll, "Error starting online processes after async initialization: %v", err)
+		base.WarnfCtx(ctx, "Error starting online processes after async initialization: %v", err)
 		atomic.CompareAndSwapUint32(&dbc.State, db.DBStarting, db.DBOffline)
+		return
```

Comment:
This is a really bad bug - is there a way that we can hit this in a unit test?

Reply:
I think I can force an error with either:
- A Bucket level TTL (https://github.com/couchbase/sync_gateway/blob/aff7f541bee8f8a3cfd26a6e10ddaa9376937e84/db/database.go#L2392C1-L2394)
- A user with only a subset of JWT properties set (https://github.com/couchbase/sync_gateway/blob/64ab69664e07f7a7108cc965a89fa74addc79d57/db/users.go#L48)

I'll see if I can stick either of these into a test (I think the user options will be more testable with Walrus/Rosmar)

Reply:
I've bashed my head trying to get "real-world" coverage (aka `RestTester`) of this all day, and it's a lot trickier than first appears. 

- I can't use bucket TTLs, as there's another (earlier) check in NewDatabaseContext that prevents the database from being created in the first place.
- I can't use invalid user config, because that's only supported in non-persistent config mode, and async online is only supported in persistent config mode.
- I can't force an error from LeakyDataStore when LastSequence is called, as persistent config means the leaky bucket wrapper gets lost.

The only remaining option I can see for a RestTester-based test is to provide a malformed OIDC callback URL, which _does_ correctly error... BUT I now have trouble hitting the codepath that runs the async `StartOnlineProcesses`, because there's an interaction between `"offline":true` and an already running "active async initialization" that I cannot hit in the right way from a test...

The only other option I have is to have a lower-level unit test only for `asyncDatabaseOnline` to make sure the database state remains Offline and doesn't revert to Online...

Reply:
I've managed to cover this line/state transition in a new test: `TestPersistentDbConfigAsyncOnlineWithInvalidConfig`